### ISSUE TITLE
Templates should use app.UseMvcWithDefaultRoute();

### ISSUE THREAD

**Issue by mlorbetske** [aspnet/templating#31]
_From @mlorbetske on August 3, 2017 17:56_

_From @Rick-Anderson on July 28, 2017 2:6_

I recommend the templates use `app.UseMvcWithDefaultRoute();` rather than

```csharp
app.UseMvc(routes =>
{
    routes.MapRoute(
        name: "default",
        template: "{controller}/{action=Index}/{id?}");
});
```

The more complex code is a barrier to entry for new users who don't need this flexibility. Advanced users will have no problem adding the above code when they need it. The code above doesn't make much sense to Razor Pages developers. Recommend this change for all templates but especially for the Razor Pages template.

cc @DamianEdwards  - part of the lower the barrier to entry effort.

_Copied from original issue: aspnet/Templates#857_

_Copied from original issue: dotnet/templating#1180_

**Comment by mlorbetske** [comment]
_From @Rick-Anderson on August 8, 2017 19:2_

The Razor Pages template should use `app.UseMvc();`

Why add controller routing to a RP project.  @rynowak  - do you agree?

**Comment by Rick-Anderson** [comment]
@rynowak  can you comment on RP routing in templates?

**Comment by rynowak** [comment]
> The Razor Pages template should use app.UseMvc();

Yes

**Comment by guardrex** [comment]
@scottaddie In addition to your #47, will you (or someone) also put one in for `UseMvcWithDefaultRoute` for the MVC template? ... It has a bearing on what they're doing with the Azure-Docs samples. I'm recommending it to them now; so if the decision is made to go that way for 2.1, they'll probably take that as an approval to go forward with `UseMvcWithDefaultRoute` today. It'll be one less thing to update later if they do.

**Comment by Rick-Anderson** [comment]
The issue title is 
**Templates should use `app.UseMvcWithDefaultRoute();`**
- with the exception of RP templates. So yes, the other templates should be changed.

**Comment by danroth27** [comment]
@Eilon 

The reason the default route is explicitly configured is because we assume that for MVC apps you will want to add and modify routes, so we want the template to give you an obvious place to do that. I think we should leave the route as is for the MVC template.

The reason the same route is present in the Web API and Razor Pages repos is because you might decide to add or scaffold MVC controllers into your app. I think this reason is less compelling, and I would be ok switching to `UseMvc()` (no convention based routes) for these templates.

**Comment by Rick-Anderson** [comment]
Historically, the routing has generated alot of confusion with customers on the [Intro to MVC tutorial](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/start-mvc?tabs=aspnetcore2x)

The default route works for many apps and probably all beginning apps. Misquoting Eilon on another issue:

> Customers that know how to use routing correctly will figure out how to add the correct code.

We could switch to  adding a comment

// See fwlinkxyz https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/routing
 `app.UseMvcWithDefaultRoute();` 

I'm less concerned about this now that RP is the new gateway drug.

**Comment by Eilon** [comment]
https://github.com/aspnet/templating/pull/47 seems to cover all that's remaining that we want to do here, after discussing with @DamianEdwards .

What we want to end up with:
* Razor Pages templates call `.UseMvc()` - that's what https://github.com/aspnet/templating/pull/47 does
* MVC templates use `.UseMvc()` and explicitly configure the default `controller/action/id` route - that's how things already are
* Web API templates use `.UseMvc()` without a default route - that's how things already are

**Comment by SteveSandersonMS** [comment]
Implemented in #47.
